viewing_party/party.py

```diff
@@ -1,23 +1,186 @@
 # ------------- WAVE 1 --------------------
 
+from tests.test_constants import GENRE_1, MOVIE_TITLE_1, RATING_1
+
+
 def create_movie(title, genre, rating):
-    pass
+    new_movie = {}
+
+
+    if title and genre and rating:
+        new_movie["title"] = title
+        new_movie["genre"] = genre
+        new_movie["rating"] = rating
+        return new_movie
+    else: 
+        return None
+
+    
+def add_to_watched(user_data, movie):
+    user_data = {"watched": []}
+    
+    movie = {
+        "title": MOVIE_TITLE_1,
+        "genre": GENRE_1,
+        "rating": RATING_1
+    }
+
+    user_data["watched"].append(movie)
+    return user_data
+
+def add_to_watchlist(user_data, movie):
+    user_data = {"watchlist": []}
+    movie = {
+    "title": MOVIE_TITLE_1,
+    "genre": GENRE_1,
+    "rating": RATING_1
+    }
+    user_data["watchlist"].append(movie)
+    return user_data
+
+def watch_movie(user_data, title):
+
+    
+    for key, value in user_data.items():
+        if key == "watchlist":
+            for i in value:
+                for key, value in i.items():
+                    if value == title:
+                        user_data["watched"].append(title)
+                        user_data["watchlist"].remove(i)
+        else:
+            return user_data
+        
+    return user_data
+
 
 # -----------------------------------------
 # ------------- WAVE 2 --------------------
 # -----------------------------------------
+def get_watched_avg_rating(user_data):
+    rating_total = []
 
+    for i in user_data["watched"]:
+        rating_total.append(i["rating"])

```

Comment:
Instead of appending the rating value to a list and then calling sum() on your list like you do in line 66, you could have rating_total initialized to 0.

Then with each iteration of the loop, you could do `rating_total += movie["rating"]

Then on line 66 you could write `return rating_total/len(rating_total)`